source/common/http/conn_manager_impl.cc

```diff
@@ -437,7 +437,7 @@ void ConnectionManagerImpl::ActiveStream::decodeHeaders(HeaderMapPtr&& headers,
   // here and keep it under 60K. Ultimately it would be nice to bring this to a lower value but
   // unclear if that is possible or not.
   if (request_headers_->byteSize() > (60 * 1024)) {
-    HeaderMapImpl headers{{Headers::get().Status, std::to_string(enumToInt(Code::BadRequest))}};
+    HeaderMapImpl headers{{Headers::get().Status, std::to_string(enumToInt(Code::RequestHeaderFieldsTooLarge))}};
```

Comment:
Ah yes, I meant to pick your brain about the 60k limit, specifically 
"Ultimately it would be nice to bring this to a lower value but unclear if that is possible or not"

We'd prefer a lower limit as well, and I wasn't sure why it was deemed implausible.  I assume lowering the default without having a config field to override it was the no-go, since some users of envoy may want to proxy large headers.  That sound right?  If so I'll update that part as well.